swan-lake/development-tutorials/source-code-dependencies/package-references.md

```diff
@@ -337,7 +337,19 @@ warning: Detected conflicting jar files:
 
 **Define the scope for a dependency**
 
-By default, the scope takes the value `default` which will add it to the final executable JAR file. If you want to restrict a certain platform dependency to be used only for testing, specify the scope as `testOnly`. This will add the platform dependncy to the test runtime but will avoid packing it into the final executable JAR file.
+By default, when the scope has not been explicitly specified for a platform dependency in the `Ballerina.toml`, it will be packaged into the final executable JAR file or the BALA file. There are two scopes that can be used to restrict this behaviour.
+
+***'TestOnly' scope***
+
+If you want to restrict a certain platform dependency to be used only for testing, specify the scope as `testOnly`. This will add the platform dependency to the test runtime but will avoid packing it into the final executable JAR file.
+
+***'provided' scope***
+
+If you want to restrict a certain platform dependency from been packed into the BALA file, specify the scope as `provided`. This will add the platform dependency to the final executable JAR file but not to the BALA file.
+
+This scope is useful in cases where the provider's license restricts the redistribution of the platform library. By specifying the "provided" scope, you ensure the dependency is accessible during both compilation and execution, yet it isn't included in the BALA package. This approach helps avoid any licensing complications associated with redistribution.
+
+In this scenario, when you use such a BALA as a dependency in another project, you'll need to explicitly specify the platform dependency in the Ballerina.toml file since it will not be included in the BALA file itself.
```

Comment:
```suggestion
In this scenario, when you use such a BALA as a dependency in another project, you would need to explicitly specify the platform dependency in the Ballerina.toml file since it will not be included in the BALA file itself.
```

Reply:
Let's also mention that provided scope is not supported with build command.